cluster-operator/src/main/java/io/strimzi/operator/cluster/model/OrderedProperties.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model;
+
+import io.strimzi.operator.cluster.InvalidConfigParameterException;
+import java.io.IOException;
+import java.io.LineNumberReader;
+import java.io.PrintWriter;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.TreeMap;
+import java.util.function.Predicate;
+
+/**
+ * A Set of ordered Name/Value pairs
+ */
+public class OrderedProperties {
+
+    private final Map<String, String> pairs;
+
+    /**
+     * The ordering of pairs when iterating the map
+     */
+    public enum Ordering {
```

Comment:
This `enum` is only abstracting the instantiation of a `Map`, and I can only foresee there being the two orderings given here, in which case I think it would be simpler and clearer to have:

```java
    public OrderedProperties(boolean keyOrdered) {
        pairs = keyOrdered ? new TreeMap<>() : new LinkedHashMap<>();
    }
```

However, the fact is that we are only ever using insertion order in the main codebase, in which case we should just have


```java
    public OrderedProperties() {
        pairs = new LinkedHashMap<>();
    }
```

Reply:
will change.